cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -524,4 +568,52 @@ protected static void countStageMetrics(StageMetrics stageMetrics,
                                           long numRecords) {
     stageMetrics.countLong(metricName, numRecords);
   }
+
+  /**
+   * This method is called when engine is present and is willing to try performing a relational transform.
+   * @param stageSpec stage specification
+   * @param transform transform plugin
+   * @param input input collections
+   * @return resulting collection or empty optional if tranform can't be done with this engine
+   */
+  public Optional<SQLEngineJob<SQLDataset>> tryRelationalTransform(StageSpec stageSpec,
+                                                          RelationalTransform transform,
+                                                          Map<String, SparkCollection<Object>> input) {
+    Map<String, Relation> inputRelations = input.entrySet().stream().collect(Collectors.toMap(
+      Map.Entry::getKey,
+      e -> sqlEngine.getRelation(new SQLRelationDefinition(e.getKey(), stageSpec.getInputSchemas().get(e.getKey())))
+    ));
+    BasicRelationalTransformContext pluginContext = new BasicRelationalTransformContext(
+      getSQLRelationalEngine(),
+      inputRelations);
+    if (!transform.transform(pluginContext)) {
+      return Optional.empty();
+    }
+    if (pluginContext.getOutputRelation() == null) {
```

Comment:
nit: Maybe add comments for these code blocks? This might make this code more understandable for someone who doesn't understand the full scope of this PR